storage/Devices/MdImpl.cc

```diff
@@ -383,8 +384,20 @@ namespace storage
 	    try
 	    {
 		const MdadmDetail& mdadm_detail = system_info.getMdadmDetail(name);
+
 		if (!mdadm_detail.devname.empty())
-		    name = DEV_MD_DIR "/" + mdadm_detail.devname;
+                {
+                    auto it = md_links.find(short_name);
```

Comment:
In the rest of the libstorage-ng code, when an iterator is declared the explicit class is always used instead of `auto`[1]. For consistency, I would use the whole class name here.

[1] Easily checked by running

```
grep -r " it =" storage/
```